### PR TITLE
bug 957802: Update docs to use docker-compose exec

### PR DESCRIPTION
**Pull request by davidflanagan**
docker-compose 1.23.0 changed the way containers are named to add a
random string of hexadecimal digits to each name, so the instructions
in docs/installation.rst that use the `docker exec -i <container_name>`
form no longer work since the container name is not predictable.

This patch updates those commands to use `docker-compose exec <service>`
instead.

The patch also updates requirements/docs.txt to keep the version of
Jinja2 in sync with what is in requires/default.txt

Test Plan:
- `cd docs; make html` and verify that the generated file looks right.
- install a new copy of Kuma, following these revised instructions

### CODE REVIEW

**Comment by davidflanagan** [comment]
Sorry to keep updating this, John. I think I'm done now.

File: docs/development.rst [hunk 1]
```diff
@@ -5,24 +5,24 @@ Development
...
+via Docker host mounting at ``/app`` within various
+Kuma containers. Useful docker sub-commands::
 
-    docker exec -it kuma_web_1 bash  # Start an interactive shell
-    docker logs kuma_web_1           # View logs from the web container
+    docker-compose exec web bash     # Start an interactive shell
+    docker-compose logs web          # View logs from the web container
     docker-compose logs -f           # Continuously view logs from all containers
-    docker restart kuma_web_1        # Force a container to reload
+    docker-compose restart seb       # Force a container to reload
```
[the remark this answers is not in the document]
Comment (davidflanagan) [reply docs/development.rst]: Oops. Thanks!

**Comment by jwhitlock** [comment]
I'm on docker-compose 1.23.1, provided by Docker for Mac, and I don't see the random slug on my containers. Weird.

Still, if some users are affected, then let's switch commands to avoid issues. Running through the new instructions.

File: docs/installation.rst [hunk 2]
```diff
@@ -153,7 +152,7 @@ Generate static assets
 Static assets such as CSS and JS are included in source form, and need to be
 compiled to their final form::
 
-    docker exec -i kuma_web_1 make build-static
+    docker-compose web make build-static
```
[the remark this answers is not in the document]
Comment (jbrandes) [reply docs/installation.rst]: Thank you! This is what I was looking for :)

**Comment by jwhitlock** [comment]
@davidflanagan I still think this change is useful, but there's some non-nits that need changing. Do you want to fix those, or should I add a commit to fix them?

**Comment by davidflanagan** [comment]
Sorry @jwhitlock. I dropped this when docker changed the docker-compose behavior back and the commands started working again as they were written... I forgot that there was other useful stuff here.

I've applied the suggestions you made (first time I've seen that Github feature!) Was that what you had in mind, or were there other changes you'd like to see before we land this?  Or should we land this as is, and if you do a follow-on PR as you mentioned above you can fine tune the edits I've made as part of that?

**Comment by davidflanagan** [comment]
Sorry @jwhitlock. I dropped this when docker changed the docker-compose behavior back and the commands started working again as they were written... I forgot that there was other useful stuff here.

I've applied the suggestions you made (first time I've seen that Github feature!) Was that what you had in mind, or were there other changes you'd like to see before we land this?  Or should we land this as is, and if you do a follow-on PR as you mentioned above you can fine tune the edits I've made as part of that?

(I wrote this comment on Monday, but then never clicked the button, so it is still here sitting in my tab on Tuesday night. Sorry to be slow to respond here!)

**Pull request merged**